src/NetMQ/NetMQPoller.cs

```diff
@@ -369,7 +429,7 @@ public Task<bool> ContainsAsync([NotNull] ISocketPollable socket)
             CheckDisposed();
 
             var tcs = new TaskCompletionSource<bool>();
-            Run(() => tcs.SetResult(m_sockets.Contains(socket)));
+            RunAsync(() => tcs.SetResult(m_sockets.Contains(socket)));
             return tcs.Task;
```

Comment:
Do you need the TCS here? Create `Task<T> RunAsync<T>(Func<T>)` and delegate to it instead.

Ditto the other `ContainsAsync` overloads.

Reply:
Again, my only change here is `Run(action)` to `RunAsync(action)`.

We *could* add the suggested `Task<T> RunAsync<T>(Func<T>)` overload... but I would striongly suggest it should be `private` or at least `internal`.

Reply:
added suggested `Task<T> RunAsync<T>(Func<T>)`